### PR TITLE
✨ Test Documentation code snippets

### PR DESCRIPTION
**Pull request by je-cook**
## Description

This is a possible addition to our CI suite that tests all code snippets in documentation and docstrings.
The documentation tests cant be run in parallel currently because each file shares a namespace.

The choice of using Sybil needs to be discussed but so far looks promising

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `pre-commit run --from-ref develop --to-ref HEAD`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,6 +34,10 @@ jobs:
           python-version: "3.10"
           bm-private-data-key: ${{ secrets.BLUEMIRA_PRIVATE_DATA_DEPLOY_KEY }}
 
+      - name: Test Documation code snippets
```
Comment (je-cook) [inline .github/workflows/ci.yml:37]:
```suggestion
      - name: Test Documentation code snippets
```

File: documentation/source/mesh/mesh.rst [hunk 2]
```diff
@@ -65,14 +69,16 @@ is integrated into bluemira.
 
 .. code-block:: python
 
+    # TODO fix
```
Comment (oliverfunk) [inline documentation/source/mesh/mesh.rst:72]: Possibly an issue?

**Review by oliverfunk — APPROVED** [review]
Great feature and getting the docs more up to scratch!

File: documentation/source/mesh/mesh.rst (hunk 2 above)
Comment (je-cook) [reply documentation/source/mesh/mesh.rst:72]: its already an issue more generally see #2999, our meshing module needs some work!

**Pull request merged**